File: 백준/Silver/10828. 스택/스택.js
const arr = require('fs').readFileSync('/dev/stdin').toString().split('\n');

const sta = [];
const ans = [];

const len = arr.shift();

for (let i = 0; i < len; i++) {        
    switch(arr[i]) {
        case 'pop': 
          ans.push(sta.pop() || -1);
          break;

        case 'size': 
          ans.push(sta.length);
          break;

        case 'empty': 
          ans.push(sta[0] ? 0 : 1);
          break;

        case 'top': 
          ans.push(sta[sta.length - 1] || -1);
          break;

        default: 
          sta.push(arr[i].split(" ")[1]);
          break;
    }
}

console.log(ans.join('\n'));

File: 백준/Silver/10828. 스택/스택.py
import sys

n, stack = int(sys.stdin.readline()), []

for _ in range(n):
    cmd = sys.stdin.readline().split()

    if cmd[0] == 'push':
        stack.append(cmd[1])
    elif cmd[0] == 'pop':
        print(stack.pop() if stack else -1)
    elif cmd[0] == 'size':
        print(len(stack))
    elif cmd[0] == 'empty':
        print(1 if not stack else 0)
    elif cmd[0] == 'top':
        print(stack[-1] if stack else -1)
